myapp.wsgi

```diff
@@ -1,54 +1,68 @@
     if not full_path.startswith(base_dir):
-        status = '403 Forbidden'
-        response = 'Access denied.'
-        response_headers = [('Content-type', 'text/plain'),
-                            ('Content-Length', str(len(response)))]
+        status = "403 Forbidden"
+        response = "Access denied."
+        response_headers = [
+            ("Content-type", "text/plain"),
+            ("Content-Length", str(len(response))),
+        ]
         start_response(status, response_headers)
-        return [response.encode('utf-8')]
-    
-    # List directory contents
-    files = []
-    directories = []
+        return [response.encode("utf-8")]
+
     try:
-        for item in os.listdir(full_path):
-            if os.path.isfile(os.path.join(full_path, item)):
-                files.append(item)
-            else:
-                directories.append(item)
-        
-        # Create JSON response
-        response = json.dumps({
-            'path': request_path,
-            'files': files,
-            'directories': directories
-        })
-
-        status = '200 OK'
-        response_headers = [('Content-type', 'application/json'),
-                            ('Content-Length', str(len(response)))]
-        start_response(status, response_headers)
-        
-        return [response.encode('utf-8')]
-    except OSError as e:
+        if os.path.isfile(full_path) and os.path.splitext(full_path)[1] == ".json":
+            status = "200 OK"
+            response_headers = [
+                ("Content-type", "application/json"),
+                ("Content-Length", str(os.path.getsize(full_path))),
+            ]
+            start_response(status, response_headers)
+            with open(full_path, "rb") as fp:
+                return iter(lambda: fp.read(65535), b"")
+        else:
+            # List directory contents
+            files = []
+            directories = []
+            for item in os.listdir(full_path):
+                if os.path.isfile(os.path.join(full_path, item)):
+                    files.append(item)
+                else:
+                    directories.append(item)
+
+            # Create JSON response
+            response = json.dumps(
+                {"path": request_path, "files": files, "directories": directories}
+            )
+
+            status = "200 OK"
+            response_headers = [
+                ("Content-type", "application/json"),
+                ("Content-Length", str(len(response))),
+            ]
+            start_response(status, response_headers)
+
+            return [response.encode("utf-8")]
+    except OSError:
         # Handle error (e.g., directory does not exist)
-        status = '404 Not Found'
-        response = 'Directory not found.'
-        response_headers = [('Content-type', 'text/plain'),
-                            ('Content-Length', str(len(response)))]
+        status = "404 Not Found"
```

Comment:
what actually changed in this file besides formatting? would have been nice to have two separate commits at least or some human description of the change

Reply:
This part is new:

https://github.com/dandi/zarr-manifests/blob/b66d8671888cfbaa8854f1c26afb354c9462cf77/myapp.wsgi#L27-L35